### DISCUSSION TITLE
WebRTC stream not as smooth on reading as RTMP stream

### DISCUSSION THREAD

**Discussion by antonymarion** [OvenMediaLabs/OvenMediaEngine#1625] [Help]
**Describe the bug**
First, please excuse my lack of knowledge about your product or if it's s a silly question :D 

**To Reproduce**
send ffmpeg stream to ovenmediaEngine over rtmp
Read webrtc stream on player

**Expected behavior**
I expect the stream to be as smooth (reading mode) as rtmp one (that I read using VLC)

**Server (please complete the following information):**
 - OS: [e.g. iOS]: all
 - OvenMediaEngine Version: [e.g. v0.10.1]: all
 - Branch: [e.g. dev]: all

**Player (please complete the following information):**
 - Device: [e.g. iPhone 8]: all
 - OS: [e.g. iOS8.1]: amm
 - Browser [e.g. stock browser, safari]: all
 - Version [e.g. 22]: amm

**Additional context**
By the way thanks for your product, since I tested both yours and srs and clearly yours does not show bad artifacts on WebRTC stream read like SRS does (using the same scenario), cf https://github.com/ossrs/srs/issues/4066

**Comment by getroot** [comment]
Does your stream contain b-frames? The browser does not support b-frames in WebRTC.
If this is not your problem, please post your encoder settings and OME log (full).

**Reply by antonymarion** [reply]
mediainfo myFile.mp4 | grep "Encoding settings" | cut -d':' -f2- | tr '/' '\n' | sed 's/ //' gives:

```
cabac=1
ref=2
deblock=1:0:0
analyse=0x3:0x113
me=hex
subme=6
psy=1
psy_rd=1.00:0.00
mixed_ref=1
me_range=16
chroma_me=1
trellis=1
8x8dct=1
cqm=0
deadzone=21,11
fast_pskip=1
chroma_qp_offset=-2
threads=17
lookahead_threads=2
sliced_threads=0
nr=0
decimate=1
interlaced=0
bluray_compat=0
constrained_intra=0
bframes=3
b_pyramid=2
b_adapt=1
b_bias=0
direct=1
weightb=1
open_gop=0
weightp=1
keyint=250
keyint_min=25
scenecut=40
intra_refresh=0
rc_lookahead=30
rc=2pass
mbtree=1
bitrate=1058
ratetol=1.0
qcomp=0.60
qpmin=0
qpmax=69
qpstep=4
cplxblur=20.0
qblur=0.5
ip_ratio=1.40
aq=1:1.00
```

**Comment by antonymarion** [comment]
Ok So you're right, it's because it contains b-frames thks
